console/src/App/AppContextProvider.js

```diff
@@ -0,0 +1,86 @@
+import React, {Component} from 'react';
+import produce from "immer";
+
+const Context = React.createContext();
+
+class AppContextProvider extends Component {
+	constructor(props){
+		super(props)
+		this.state = {
+			userId: 1,
+			userSettings: {
+				color: '#FFFFFF',
+				fontSize: 0,
+				skin: props.defaultSkin || 'carbon',
+			},
+			updateAppState: this.updateAppState,
```

Comment:
Running these two functions through state (and therefore immer) is probably not ideal since consumers could inadvertently overwrite them.